storm-server/src/main/java/org/apache/storm/daemon/supervisor/SupervisorUtils.java

```diff
@@ -145,6 +146,18 @@ protected LSWorkerHeartbeat readWorkerHeartbeatImpl(Map<String, Object> conf, St
     }
 
     private boolean isWorkerHbTimedOutImpl(int now, LSWorkerHeartbeat whb, Map<String, Object> conf) {
-        return (now - whb.get_time_secs()) > ObjectReader.getInt(conf.get(Config.SUPERVISOR_WORKER_TIMEOUT_SECS));
+        Integer timeoutSecs;
+        if (conf.containsKey(Config.TOPOLOGY_WORKER_TIMEOUT_SECS)) {
```

Comment:
Will there ever be a time when you get the topology conf?

Reply:
Current code always passes Storm config, but I want to handle the case where a topology config is passed.

Reply:
In which case will a `TOPOLOGY_WORKER_TIMEOUT_SECS` be passed here